src/extension/layer-filters/src/main/java/au/org/emii/geoserver/extensions/filters/LayerFiltersService.java

```diff
@@ -133,6 +133,11 @@ public void enabledFilters(HttpServletRequest request, HttpServletResponse respo
         String workspace = request.getParameter("workspace");
         String layer = request.getParameter("layer");
 
+        if(workspace==null) {
+            workspace = layer.split(":")[0];
```

Comment:
Can it at least fall over at this point with a handled exception? ;)

Looks like in that file only 'enabledFilters' and 'uniqueValues' are actually dealing with URL parameters via the HttpServletRequest, so could it not have a common method of splitting the workspace from the URL and failing cleanly if a) there is a missing workspace parameter AND b) the layer name doesn't contain the workspace, and then sending a coherent HTTP response instead of just remaining unhandled? Doesn't seem like too much effort if it's only in that one file!